src/ariel-os-storage/src/storage.rs

```diff
@@ -124,10 +102,10 @@ impl<F: NorFlash> Storage<F> {
     }
 
     /// Resets the flash in the entire flash range of this [`Storage`] instance.
-    pub async fn erase_all(
-        &mut self,
-    ) -> Result<(), sequential_storage::Error<<F as ErrorType>::Error>> {
-        erase_all(&mut self.flash, self.storage_range.clone()).await
+    pub async fn erase_all(&mut self) -> Result<(), Error> {
+        erase_all(&mut self.flash, self.storage_range.clone())
+            .await
+            .map_err(Error::from)
```

Comment:
optional: I usually write this this way (`?` already uses `From`); I'm not sure it results in the same binary however. This is really just a suggestion (the other occurrences would need to be updated as well if you went with this):

```suggestion
        Ok(erase_all(&mut self.flash, self.storage_range.clone()).await?)
```

Reply:
I used your suggestion in a previous version of this PR and you` suggested `map_err` [here](https://github.com/ariel-os/ariel-os/pull/836#discussion_r1959936211) :laughing:

Reply:
Yes, but to map/wrap upstream errors into specific custom errors :slightly_smiling_face: